### PR TITLE
Extend BaseRenderer Docs

### PR DESCRIPTION
**Pull request by EdoPut**
I added a little more description of the BaseRenderer process and methods

### CODE REVIEW

**Comment by nemesisdesign** [comment]
@EdoPut any progress on this?

**Comment by EdoPut** [comment]
Not until I get time to review it and rewrite what is needed

2017-04-28 10:08 GMT+02:00 Federico Capoano <notifications@github.com>:

File: netjsonconfig/backends/base.py [hunk 1]
```diff
@@ -107,20 +121,35 @@ def render(self, files=True):
...
-            # to separate blocks
+            # to separate blocks of configuration
+            # generated by different renderers
             if output and additional_output:
                 output += '\n'
+
+            # concatenate the render configuration
             output += additional_output
+
+        # are we required to include
```
Comment (EdoPut) [inline netjsonconfig/backends/base.py]: As this is the base renderer and it's process of rendering configuration is not described elsewhere I would like to mantain this comment.

File: netjsonconfig/backends/base.py [hunk 2]
```diff
@@ -51,21 +53,26 @@ def _load(self, config):
...
         if not isinstance(templates, list):
             raise TypeError('templates argument must be an instance of list')
         # merge any present template with main configuration
         base_config = {}
+
         for template in templates:
             template = self._load(template)
-            base_config = merge_config(base_config, template)
-        if base_config:
+        base_config = merge_config(base_config, template)
```
Comment (nemesisdesign) [inline netjsonconfig/backends/base.py]: something wrong happened in these 2 lines

File: netjsonconfig/backends/base.py [hunk 3]
```diff
@@ -107,20 +121,35 @@ def render(self, files=True):
...
+            # concatenate the render configuration
             output += additional_output
+
+        # are we required to include
+        # additional files?
         if files:
-            # render files
+            # render additional files
             files_output = self._render_files()
+
```
Comment (nemesisdesign) [inline netjsonconfig/backends/base.py]: remember to remove this blank line please

File: netjsonconfig/backends/base.py [hunk 4]
```diff
@@ -256,7 +277,10 @@ def render(self):
 
     def get_context(self):
         """
-        Builds context that is passed to jinja2 templates
+        Builds context dictionary to be used in jinja2 templates
+        For every getter method suche as ``_get_name`` creates an entry in
```
Comment (nemesisdesign) [inline netjsonconfig/backends/base.py]: `suche` should be `such`

`getter` and `setter` methods are a slightly different thing in python (see [descriptor pattern](https://docs.python.org/3/howto/descriptor.html) for more information).

Something like the following would be more appropriate:
"For every method prefixed with ``_get_``, such as ``_get_name``, ..."

**Comment by nemesisdesign** [comment]
ok @EdoPut, now please squash the commits into one so I can merge

**Pull request merged**